internal/webhook/validate_with_warn.go

```diff
@@ -0,0 +1,49 @@
+package webhook
+
+import (
+	"fmt"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func validateWarningRules(response *admissionv1.AdmissionResponse, err *httpErr, request *checkRequest, checkers ...checker) (*admissionv1.AdmissionResponse, *httpErr) {
+	//If the request is rejected during rule checking, then do not process the warning rules.
+	if !response.Allowed {
+		return response, err
+	}
+
+	msg := make([]string, 0)
+	// Combine all warining msg together
+	for _, c := range checkers {
+		resp, _ := c.check(request)
+		if len(resp.Warnings) > 0 {
+			msg = append(msg, resp.Warnings...)
+		}
+	}
+
+	if wariningResponseCount := len(msg); wariningResponseCount == 0 {
```

Comment:
We can avoid variable resource allocation if we do not need the `wariningResponseCount` var.

Reply:
It enhances the readability of the code.